src/collector/worker.rs

```diff
@@ -0,0 +1,139 @@
+// This software is licensed under a dual license model:
+//
+// GNU Affero General Public License v3 (AGPLv3): You may use, modify, and
+// distribute this software under the terms of the AGPLv3.
+//
+// Elastic License v2 (ELv2): You may also use, modify, and distribute this
+// software under the Elastic License v2, which has specific restrictions.
+//
+// We welcome any commercial collaboration or support. For inquiries
+// regarding the licenses, please contact us at:
+// vectorchord-inquiry@tensorchord.ai
+//
+// Copyright (c) 2025 TensorChord Inc.
+use crate::collector::types::{PGLockGuard, Result};
+use std::cell::OnceCell;
+use std::ffi::{CStr, CString};
+use std::fs;
+use std::path::Path;
+
+const SQLITE_DATABASE: Option<&CStr> = Some(c"main");
+const COLLECTOR_DIR: &str = "pg_vchord_sample";
+const COLLECTOR_VERSION_PATH: &str = "pg_vchord_sample/VERSION";
+const COLLECTOR_VERSION: u32 = 1;
+
+const MULTI_ACCESS_LOCK: u32 = 0;
+
+thread_local! {
+    static GLOBAL_INIT: OnceCell<bool> = const { OnceCell::new() };
+}
+
+fn init() -> Result<bool> {
+    let path = Path::new(COLLECTOR_VERSION_PATH);
+    if path.exists() && path.is_file() {
+        let content = fs::read_to_string(path)?;
+        if content.trim().parse::<u32>()? == COLLECTOR_VERSION {
+            return Ok(true);
+        }
+    }
+    match fs::remove_dir_all(COLLECTOR_DIR) {
+        Ok(_) => (),
+        Err(e) if e.kind() == std::io::ErrorKind::NotFound => (),
+        Err(e) => return Err(e.into()),
+    };
+    fs::create_dir_all(COLLECTOR_DIR)?;
+    fs::write(path, format!("{COLLECTOR_VERSION}"))?;
+    Ok(true)
+}
+
+fn sqlite_connect(
+    create: bool,
+    database_oid: u32,
+    index_oid: u32,
+) -> Result<Option<rusqlite::Connection>> {
```

Comment:
Why `sqlite_connect` is associated with `index_oid`? Why not keep the connection alive? Why put the logic here, not where the samples are written?